frappe/public/js/frappe/model/create_new.js

```diff
@@ -292,7 +292,7 @@ $.extend(frappe.model, {
 			opts.source_name = opts.frm.doc.name;
 
 		// Allow opening a mapped doc without a source document name
-		} else if (!opts.frm) {
+		} else if (!opts.frm && !opts.source_name) {
 			opts.source_name = null;
 		}
```

Comment:
```suggestion
		}
```
@casesolved-co-uk Removing this entire code block will have the same effect.

Also, that code was introduced via [this PR](https://github.com/frappe/frappe/pull/4033). Make sure if the case that the PR was trying to resolve is also handled.

Reply:
It is clear that my change will have no effect.

However it is not my duty to check that your new proposed change is valid.